app/models/appointment_slot_preset.rb

```diff
@@ -1,8 +1,29 @@
 class AppointmentSlotPreset < ActiveRecord::Base
+  include PushUpdatesMinimal
+
+  validate :no_conflict, on: :update
+
   validates :day, numericality: { only_integer: true, greater_than: 0, less_than: 8  }
   validates :hours, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than: 24 }
   validates :minutes, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than: 60 }
   validates :quota, numericality: { only_integer: true, greater_than_or_equal_to: 1 }
 
   scope :ascending, -> { order('hours ASC').order('minutes ASC') }
+
+  after_save :push_changes
+  after_destroy :push_changes
+  push_targets [ Channel::STOCK_CHANNEL ]
+
+  private
+
+  def no_conflict
+    conflicts = AppointmentSlotPreset
+      .where({ day: self.day, hours: self.hours, minutes: self.minutes })
```

Comment:
Do we need to write `self` here? I think writing `day`, `hours` and `minutes` will work on object. Let me know if it's something different.

Reply:
It's just a JS habit, of typing `this.` 
I do value codes that are explicit generally though, I really don't mind a few extra characters for more clarity

Reply:
https://www.rubydoc.info/gems/rubocop/0.16.0/Rubocop/Cop/Style/RedundantSelf

Actually writing such self is not good practice in the ruby world so just thought to suggest this change.

Reply:
Interesting, I'm not sure I agree with Rubocop though, different mindsets I guess.
I'll change it to be consistent
Thanks for sharing 😄